scripts/refactor_testing_query_courseid.py

```diff
@@ -0,0 +1,176 @@
+        page_size (int, optional): The page size. Defaults to 100.
+    
+    Returns:
+        str: JSON string with extracted information or error message.
+    """
+    class_info, error_message = validate_class_name(class_name)
+    if error_message:
+        return json.dumps(error_message, indent=4)
+
+    subject_area_code, number = class_info
+
+    base_url = "https://gateway.api.berkeley.edu/uat/sis/v1/classes?"
+    url_params = f"term-id={term_id}&subject-area-code={subject_area_code}&catalog-number={number}&page-number={page_number}&page-size={page_size}"
+    full_url = base_url + url_params
+
+    headers = {
+        "app_id": os.getenv('APP_ID'),
+        "app_key": os.getenv('APP_KEY'),
+    }
+
+    response = requests.get(full_url, headers=headers)
+
+    if response.status_code == 200:
+        data = response.json()['apiResponse']['response']
+        classes = data['classes']
+        extracted_info = {
+            "display_name": classes[0]['displayName'],
+            "department": classes[0]['course']['subjectArea']['description'],
+            "enrollment_count": sum(section['aggregateEnrollmentStatus']['enrolledCount'] for section in classes)
+        }
+        return json.dumps(extracted_info, indent=4)
+    
+    if check_pattern(class_name):
+        desired_display_name = extract_pattern(class_name)
+        if not desired_display_name:
+            error_string = f"Class name {class_name} is invalid. Please enter a valid class name."
+            return json.dumps(error_string, indent=4)
+        
+        subject_area_code, number = desired_display_name.split()
+        full_url = base_url + f"term-id={term_id}&subject-area-code={subject_area_code}&catalog-number={number}&page-number={page_number}&page-size={page_size}"
+
+        response = requests.get(full_url, headers=headers)
+        if response.status_code == 200:
+            data = response.json()['apiResponse']['response']
+            classes = data['classes']
+            extracted_info = {
+                "display_name": classes[0]['displayName'],
+                "department": classes[0]['course']['subjectArea']['description'],
+                "enrollment_count": sum(section['aggregateEnrollmentStatus']['enrolledCount'] for section in classes)
+            }
+            return json.dumps(extracted_info, indent=4)
+    
+    error_string = f"Failed to retrieve data for {class_name}. Status code: {response.status_code}"
+    return json.dumps(error_string, indent=4)
+
+def main():
+    """
+    Main function to handle command-line arguments or interactive input.
+    """
+    parser = argparse.ArgumentParser(description="Fetch course information based on term ID and class name.")
+    parser.add_argument('term_id', type=str, nargs='?', help="The term ID (e.g., '2232').")
+    parser.add_argument('class_names', type=str, nargs='*', help="One or more class names (e.g., 'data8', 'compsci189').")
+    
+    args = parser.parse_args()
+    
```

Comment:
i would also add some logic to check if the proper env vars are set before continuing script execution.